avivator/src/components/Controller/components/VolumeButton.js

```diff
@@ -89,12 +89,18 @@ function VolumeButton() {
           // eslint-disable-next-line no-unused-expressions
           if (use3d) {
             toggleUse3d();
+            setViewerState({
+              isChannelLoading: selections.map(_ => true)
+            });
             getMultiSelectionStats({ loader, selections, use3d: !use3d }).then(
               ({ domains, sliders }) => {
                 setPropertiesForChannels(range(selections.length), {
                   domains,
                   sliders
                 });
+                setViewerState({
+                  isChannelLoading: selections.map(_ => false)
+                });
               }
```

Comment:
Another example of where getting stats independently would allow the UI to update on demand rather than waiting for each channel before any UI update.